docs/source/app-dev/services.rst

```diff
@@ -59,6 +59,23 @@ Commands can be labeled with two application-specific IDs, both of which are ret
 
 For full details, see :ref:`the proto documentation for the service <com.digitalasset.ledger.api.v1.CommandSubmissionService>`.
 
+.. _command-submission-service-deduplication:
+
+Command deduplication
+---------------------
+
+The command submission service deduplicates submitted commands:
+
+- Applications can assign a "time to live" (TTL) value to each command.
+- Duplicate command submissions (using the same submitter and command ID) will be ignored until either the TTL of the original command has elapsed or the original submission was rejected, whichever comes first.
+- Command deduplication is only guaranteed to work if all commands are submitted to the same participant. On some ledgers, the deduplication may also work across participants.
+- A command submission will return:
+  - The result of the submission, if the command was submitted for the first time
```

Comment:
the result of a submission is returned via command completion service and not by command submission service, unless this section is actually talking about the command service. we may also get an OK status, but a duplicate command error on the command completion service for ledgers that also deduplicate outside participants.

Reply:
Think of the result of a submission in terms of [`SubmissionResult`](https://github.com/digital-asset/daml/blob/master/ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/SubmissionResult.scala).

Reply:
This is talking about the submission service - it returns either an empty response (if the command was successfully submitted), or an error (e.g., if there were command interpretation errors).

Deduplicated submissions will return the same response as the original submission, if that information is available.